app/src/components/pages/ProfessorProfile.tsx

```diff
@@ -1,7 +1,165 @@
-import React from 'react';
+import { BubbleChart, Code, DataObject, DeveloperBoard } from '@mui/icons-material';
+import { Avatar, ButtonBase, Card, Grid, ListItem, ListItemAvatar, ListItemText, Typography } from '@mui/material';
+import { Faculty, Role } from 'components/shared/constants/timetable.constants';
+import { ICourse, IProfessor, IProfessorPreference } from 'components/shared/interfaces/timetable.interfaces';
+import React, { ReactElement } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { getCoursesForProfessor } from 'utils/utils';
+import classData from 'data/clean.json';
 
 function ProfessorProfile() {
-  return <div>ProfessorProfile</div>;
+  const navigate = useNavigate();
+
+  const professor: IProfessor = {
+    active: true,
+    id: 123,
+    role: Role.User,
+    username: 'hausi',
+    preferences: [
+      { id: { subject: 'SENG', code: '265', term: 'SPRING' }, preference: 172 },
+      { id: { subject: 'SENG', code: '480D', term: 'SPRING' }, preference: 150 },
+      { id: { subject: 'SENG', code: '265', term: 'SUMMER' }, preference: 133 }
+    ],
+    faculty: 'SENG'
+  };
+
+  const currentlyTeaching: ICourse[] = getCoursesForProfessor(
+    professor.username,
+    JSON.parse(JSON.stringify(classData))
+  );
+
+  function getIcon(faculty: Faculty): ReactElement {
+    switch (faculty) {
+      case 'SENG':
+        return <Code />;
+      case 'CSC':
+        return <BubbleChart />;
+      case 'ECE':
+        return <DeveloperBoard />;
+      default:
+        return <DataObject />;
+    }
+  }
+
+  function getBgColor(faculty: Faculty): string {
+    switch (faculty) {
+      case 'SENG':

```

Comment:
Maybe inline this with a dict? Kind of a nitpick though

Reply:
Good idea.